testing/ember-app/tests/utils/function-resource/js-test.ts

```diff
@@ -256,4 +259,37 @@ module('Utils | resource | js', function (hooks) {
       assert.verifySteps(['resolved 1', 'resolved 2', 'resolved 3']);
     });
   });
+
+  module('with owner', function (hooks) {
+    class TestService extends Service {
+      @tracked count = 1;
+    }
+
+    class Test {
+      // @use is required if a primitive is returned
+      @use data = resource(() => {
+        const test = (getOwner(this) as Owner).lookup('service:test') as TestService;
```

Comment:
the `this` here is the `Test` instance -- is this intended?

Reply:
that idea I followed was this:

- Function passed into `resource` is bound to `null` (somewhat to "neutralize" I guess?)
- Use `setOwner()` to set the owner to the context of the function (`null`)...
- .. to then use `getOwner(this)` within the function
- .. to get the owner of to which the function was bound

now you made me thinking 🤔 hmm... so my usage of `getOwner(this)` is perhaps not the one I intended to do? What would be the correct parameter to `getOwner()` then ?

Reply:
> What would be the correct parameter to getOwner() then ?

```js
let myResource = resource(stuff => {
  let owner = getOwner(stuff);
  
  stuff.on.cleanup(() => { ... }
  
  return ...
});
```

perhaps, in the tests, define the resource outside the scope of a class? that would ensure that we getOwner on the correct thing

Reply:
Hmm... I am skeptical about this. The whole idea of is passing hooks/dependencies into the resource. The `owner` would be part of these dependencies, but you'd make it accessible through a totally different API - that doesn't make sense to me (and also is an invisible API). I see these two implementations as of now:

1. Use `setOwner()` to register it with the function (my current impl of this PR, would need to check the impl once more); the idea is to use `getOwner(this)`, as this is "commonly" used in ember.
2. Put it onto hooks (my original idea of this PR 😅)
3. ... the option I don't see 🤷 

The idea is to create an API that is _expectable_.

Reply:
I think I'm leaning towards the original idea :upside_down_face: because:
```js
resource(({ on, owner }) => {
  seems nicer?
})
```

Reply:
ok ok. We are back 😅